scripts/GFF_to_XGDB_Standard.pl

```diff
@@ -1,307 +1,306 @@
+                        }
+                    }
+                  }
+                  else {
+                    $CDSstop  = $sortedCDSarray[0];
+                    $CDSstart = $sortedCDSarray[$#sortedCDSarray];
+                    if ( $CDSstop eq $l_pos ) {
+                        if ( $CDSstart eq $r_pos ) {
+                            $gene_structure =~ s/(\d+)$/\&gt\;$1/;
+                            $gene_structure =
+                              "complement(join(&lt;" . $gene_structure . "))";
+                        }
+                        else {
+                            $gene_structure =
+                              "complement(join(&lt;" . $gene_structure . "))";
+                        }
+                    }
+                    else {
+                        if ( $CDSstart eq $r_pos ) {
+                            $gene_structure =~ s/(\d+)$/\&gt\;$1/;
+                            $gene_structure =
+                              "complement(join(" . $gene_structure . "))";
+                        }
+                        else {
+                            $gene_structure =
+                              "complement(join(" . $gene_structure . "))";
+                        }
+                    }
+                }
+              }
+              else {
+                $CDSstart = 0;
+                $CDSstop  = 0;
+              }
+              print
+"INSERT into $table (geneId,gseg_gi,strand,l_pos,r_pos,gene_structure,description,note,CDSstart,CDSstop,transcript_id,locus_id) VALUES ('$geneId','$chr','$strand',$l_pos,$r_pos,'$gene_structure','$description','$note',$CDSstart,$CDSstop,'$transcript_id','$locus_id');\n";
+              $gene_structure = "";
+              @CDSarray       = ();
+              @CDS_UTRarray   = ();
+              @CDS_Exonarray  = ();
+              if ($flagALLDONE) { next FILELOOP; }
+            }
+
+            # Set flagNewGene to 1 (the previous gene having been processed)
+            $flagNewGene = 1;
+            $strand      = $tmpstrand;
+            $l_pos       = $tmpl_pos;
+            $r_pos       = $tmpr_pos;
+
+            if (   $attributes =~ /ID=(\S+?);/i
+                || $attributes =~ /mRNA\s+([0-9]{5}\.m[0-9]{6});/i )
+            {
+                $transcript_id = $1;
+                $description   = $attributes;
+                $description =~ s/\s+$//g;
+                $description =~ s/"/ /g;
+                $description =~ s/'/prime/g;
+                $note = $description;
+            }
+            $geneType = $type;
+            $geneId   = $transcript_id;
+            if (   $attributes =~ /Parent=(\S+?);/
+                || $attributes =~ /Parent=(\S+)/ )
+            {
+                $locus_id = $1;
+            }
+            else {
+                $locus_id = $geneId;
+            }
+            $chr = $tmpchr;
+        }
+        elsif ($type =~ m/UTR/i) {
```

Comment:
This approach—parsing out the third GFF3 value and testing it for `UTR` inclusion—is much safer. If we wanted to be *really* strict, we could test whether `$type` was one of the several feature types officially described and sanctioned by the Sequence Ontology, but I think in this case the simple and concise solution is better.